packages/dev/src/cli.js

```diff
@@ -0,0 +1,68 @@
+import { Command } from 'commander';
+
+import { config } from './commands/config.js';
+import { init } from './commands/init.js';
+import { setInstance } from './commands/set-instance.js';
+import { setRoot } from './commands/set-root.js';
+import { setup } from './commands/setup.js';
+import { watch } from './commands/watch.js';
+
+export default function cli() {
+  const program = new Command();
+
+  program.name('clerk-dev').description('CLI to make developing Clerk packages easier').version('0.0.0');
+
+  program
+    .command('init')
+    .description('Perform initial one-time machine setup')
+    .action(async () => {
+      await init();
+    });
+
+  program
+    .command('config')
+    .description('Open the clerk-dev config file in EDITOR or VISUAL')
+    .action(async () => {
+      await config();
+    });
+
+  program
+    .command('set-root')
+    .description(
+      'Set the location of your checkout of the clerk/javascript repository to the current working directory',
+    )
+    .action(async () => {
+      await setRoot();
+    });
+
+  program
+    .command('set-instance')
+    .description('Set the active instance to the provided instance name')
+    .argument('<name>', 'name of instance listed in dev.json')
+    .action(async instance => {
+      await setInstance({ instance });
+    });
+
+  program
+    .command('setup')
+    .description(
+      'Install the monorepo versions of Clerk packages listed in the package.json file and perform framework configuration for the current working directory',
+    )
+    .option('--no-js', 'do not customize the clerkJSUrl')
+    .action(async ({ js }) => {
+      await setup({ js });
+    });
+
+  program
+    .command('watch')
+    .description(
+      'Start the dev tasks for all Clerk packages listed in the package.json file in the current working directory, including clerk-js (unless --no-js is specified)',
+    )
+    .option('--js', 'only start the watcher for clerk-js')
+    .option('--no-js', 'do not spawn the clerk-js watcher (macOS only)')
```

Comment:
These two boolean flags feel a bit awkward, what happens if in the future one wants to not spawn other things?

How about replacing this with an `--ignore` flag that accepts multiple values?

Then the default is that clerk-js is built, too, and one can use `--ignore` to not build it or other stuff

Reply:
With `--no-js`, we can align with the flags passed to `clerk-dev setup`. So for example, `clerk-dev setup --no-js` would be followed by `clerk-dev watch --no-js`. On Linux you could do `clerk-dev setup --no-js` followed by `clerk-dev watch` since `watch` doesn't spawn the `clerk-js` builder on Linux. I can definitely see the awkwardness when you factor in the Linux implementation, but given our entire team uses macOS I felt that was acceptable. Is there another flag that would be less awkward but retain the alignment with `clerk-dev setup` so that users don't need to learn two different flags?

I think the `--ignore` flag is a good idea, but is something I'd like to save for later once we have a better idea of what situations would cause a developer to want to ignore other packages.